File: Sarvam C-DAC Project/Sarvam C-DAC Spring Boot/Sarvam/src/main/java/com/cdac/sarvam/dto/MasterModuleDTO.java
package com.cdac.sarvam.dto;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MasterModuleDTO {
    private Integer moduleId;
    private String moduleName;
    private LocalDateTime moduleStartDate;
    private LocalDateTime moduleEndDate;
    private Long instructorId;
    private String instructorName;
    private String instructorLastName;
    private Long moCoId;
    private String moCoName;
    private String lastName;
    private Byte noOfDays;
    private Byte durationHours;
    private List<MasterModuleSubpointDTO> masterModuleSubpoints;
}

File: MS.NET/Sarvam Backend/DTO/MasterModuleDTO.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Sarvam.DTO
{
    public class MasterModuleDTO
    {
        public int ModuleId { get; set; }
        public string ModuleName { get; set; }
        public DateTime ModuleStartDate { get; set; }
        public DateTime ModuleEndDate { get; set; }
        public long InstructorId { get; set; }
        public string? InstructorName { get; set; }
        public string? InstructorLastName { get; set; }
        public long MoCoId { get; set; }
        public string? MoCoName { get; set; }
        public string? LastName { get; set; }
        public byte NoOfDays { get; set; }
        public byte DurationHours { get; set; }

        [NotMapped]
        public ICollection<MasterModuleSubpointDTO>? MasterModuleSubpoints { get; set; }

    }
}
